MyStore/src/GeneratorData.cpp

```diff
@@ -0,0 +1,35 @@
+#include "GeneratorData.h"
+
+const int period_for_element_milliseconds = 1;
+
+
+std::vector<InfoProduct> GeneratorData::DoGenarate()
+{
+	InfoProduct data;
+
+    for (int i = 0; i < m_colors.size(); i++)
+	{
+		if (m_data.size() == m_max_amount_data)
+			break;
+	    data.m_product_color = m_colors[i];
+
+		for (int j = 0; j < m_names.size(); j++)
+	{
+			if (m_data.size() == m_max_amount_data)
+				break;
+			data.m_product_name = m_names[j];
+		    m_data.push_back(data);
+			std::this_thread::sleep_for(std::chrono::milliseconds(period_for_element_milliseconds));
+		}
+	}				
+			return m_data;
+}
+
+bool GeneratorData::ShowData()
+{
+	for (int i = 0; i < m_data.size(); i++)
+	{
+		std::cout << "Color: " << m_data[i].m_product_color << "  " << "Name: " << m_data[i].m_product_name << '\n';
+	}

```

Comment:
To avoid code duplication separate product output from generator and from parser.